src/saving/serializers/ThriveTypeConverter.cs

```diff
@@ -30,7 +30,7 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
     {
         if (destinationType == typeof(string))
         {
-            return ThriveJsonConverter.Instance.SerializeObject(value);
+            return ThriveJsonConverter.Instance.SerializeObject(value, destinationType);
```

Comment:
Please add a comment also here saying that this is pretty weird that we need to go "here's value, and it's type is string, even though it isn't"
Probably when we say that value's type is string, the TypeNameHandling.auto then realizes that value's type is not string, so it writes out the dynamic type in that case.
This is the custom serializer type writing out code that doesn't trigger correctly: https://github.com/Revolutionary-Games/Thrive/blob/834143d0ab66acf8fc65f9a642bd77fd8c16eb5b/src/saving/serializers/ThriveJsonConverter.cs#L516

Here's what the speciesinpatch correct generated json looks like (when there is an extinct species):
```
                "SpeciesInPatch": {
                  "{\"$ref\":\"3\"}": 1547,
                  "{\"$ref\":\"5\"}": 484,
                  "{\"$ref\":\"83\"}": 576,
                  "{\"$ref\":\"86\"}": 408,
                  "{\"$ref\":\"36\"}": 227,
                  "{\"$id\":\"101\",\"$type\":\"MicrobeSpecies, Thrive\",\"IsBacteria\":true,\"MembraneType\":\"silica\",\"MembraneRigidity\":0.21,\"InitialCompounds\":{\"atp\":30.0,\"glucose\":10.0},\"Genus\":\"Antoxnea\",\"Epithet\":\"Okictus\",\"Colour\":{\"r\":0.690856636,\"g\":0.840199649,\"b\":0.7115342,\"a\":0.9585607},\"Aggression\":91.01558,\"Opportunism\":110.051453,\"Fear\":91.58104,\"Activity\":11.5477524,\"Focus\":0.0,\"Population\":1,\"Generation\":1,\"Organelles\":{\"Organelles\":[{\"$id\":\"102\",\"Definition\":\"cytoplasm\",\"PlacedThisSession\":false,\"Position\":{\"Q\":0,\"R\":0},\"Orientation\":0}],\"onAdded\":null,\"onRemoved\":null},\"ID\":31,\"PlayerSpecies\":false}": 0,
                  "{\"$ref\":\"75\"}": 271,
                  "{\"$ref\":\"21\"}": 154,
                  "{\"$ref\":\"16\"}": 479
                },
```
So by removing this destinationType from here and investigating why the `\"$type\":\"MicrobeSpecies...` part doesn't get included is how to start on the issue.

(those last few parts are for including extra info in the new issue)

Reply:
Thinking about this more, perhaps the fix is as simple as detecting that `value` has JSONAlwaysDynamicType on it, and then getting the type's base class, and passing that as the second argument. Then the type of `value` wouldn't match the given type. Or another approach could be to somehow make sure that `[JSONAlwaysDynamicType]` works when added to MicrobeSpecies, but that then would force dynamic type usage in some cases where objects directly hold MicrobeSpecies and don't really want to use dynamic types. So overall I think the first solution I suggested in this comment should be tried.

Reply:
One more thing I forgot to say. Because when stringifying a dictionary key, the converted object is the top level object, and it doesn't make much sense for that to use dynamic type by default, so by giving another Type to the JSON lib, we are in effect saying that, "I think the top level type is X" then the JSON library goes "hang on that's not correct, so you seem to be using dynamic types, let me write that $type out for you". At least that's my reasoning why my solution might work, and that would explain this whole ordeal why we are having this problem in the first place.

If that solves it, then styleguide or some help documentation should be maybe updated with this information in case someone runs into it in the future.

Reply:
Agreed, I think that's a logical explanation why the serializer behaves the way it is. Btw I tried your first solution with passing the type's base class, and it looks like to be working just fine. This is the code that I use from your suggestion:
![code4](https://user-images.githubusercontent.com/54026083/106120793-66521000-6189-11eb-92e4-6f6721e2821b.png)
Do we still need to open a new issue or just write comments there explaining why such a code is used?

Reply:
No need to open a separate issue if you can put that fix on this branch. Btw I think you typoed that last line.
Perhaps something like:
```
var type = value.GetType();
if(type.CustomAttributes... and so on){
    type = type.BaseType;
}
return ThriveJsonConverter.Instance.SerializeObject(value, type);
```